crates/librqbit/src/torrent_state/live/mod.rs

```diff
@@ -1013,22 +1013,37 @@ impl PeerHandler {
         if self.state.is_finished_and_no_active_streams() {
             debug!("torrent finished, not re-queueing");
             pe.value_mut().state.set(PeerState::NotNeeded, &pstats);
+            // also cancel all retried dead peers
+            if ! self.state.cancellation_token.is_cancelled() {
```

Comment:
a few problems with this:
- this will cancel all tasks, not just peers. This is major and needs to be fixed
- you don't need to check is_cancelled() to cancel.
- I'd rather not use the cancellation token for this as it's specifically for destroying everything + it's a one-time thing, it's not reusable, i.e. you can't reinstate it after canceling. But a torrent being finished is not "the end". The user might change "only_files" for example.

In general I think these delayed reconnect attempts aren't harmful, especially when it will be done only once, and it's easier to keep it like it is

Reply:
I see,  
it was rather a test - to see if  scheduled retries are canceled - and  yes they are.
But as you said - much more is canceled too - for instance incoming peers are stop being served. Is actually same as pausing torrent.

Delayed reconnect are no problem as you stated,  I was just curious why they are appearing - even several hours after torrent is finished. It's rather small annoyance in log.  

Maybe to create a child cancellation token just for this purpose?